jetty-websocket/jetty-websocket-tests/src/test/java/org/eclipse/jetty/websocket/tests/client/WebSocketClientTest.java

```diff
@@ -171,6 +172,76 @@ public void testBasicEcho_FromClient() throws Exception
+    {
+        CloseTrackingEndpoint cliSock = new CloseTrackingEndpoint();
+
+        client.getPolicy().setIdleTimeout(10000);
+
+        URI wsUri = WSURI.toWebsocket(server.getURI().resolve("/echo"));
+        ClientUpgradeRequest request = new ClientUpgradeRequest();
+        request.setSubProtocols("echo");
+        Future<Session> future = client.connect(cliSock, wsUri, request);
+
+        try (Session sess = future.get(30, TimeUnit.SECONDS))
+        {
+            assertThat("Session", sess, notNullValue());
+            assertThat("Session.open", sess.isOpen(), is(true));
+            assertThat("Session.upgradeRequest", sess.getUpgradeRequest(), notNullValue());
+            assertThat("Session.upgradeResponse", sess.getUpgradeResponse(), notNullValue());
+
+            Collection<WebSocketSession> sessions = client.getOpenSessions();
+            assertThat("client.sessions.size", sessions.size(), is(1));
+
+            RemoteEndpoint remote = cliSock.getSession().getRemote();
+            remote.sendPartialString("Hello", false);
+            remote.sendPartialString(" ", false);
+            remote.sendPartialString("World", true);
+
+            // wait for response from server
+            String received = cliSock.messageQueue.poll(5, TimeUnit.SECONDS);
+            assertThat("Message", received, containsString("Hello World"));
+        }
+    }
+
+    @Test
+    public void testBasicEcho_PartialText_WithPartialBinary_FromClient() throws Exception
+    {
+        CloseTrackingEndpoint cliSock = new CloseTrackingEndpoint();
+
+        client.getPolicy().setIdleTimeout(10000);
+
+        URI wsUri = WSURI.toWebsocket(server.getURI().resolve("/echo"));
+        ClientUpgradeRequest request = new ClientUpgradeRequest();
+        request.setSubProtocols("echo");
+        Future<Session> future = client.connect(cliSock, wsUri, request);
+
+        try (Session sess = future.get(30, TimeUnit.SECONDS))
+        {
+            assertThat("Session", sess, notNullValue());
+            assertThat("Session.open", sess.isOpen(), is(true));
+            assertThat("Session.upgradeRequest", sess.getUpgradeRequest(), notNullValue());
+            assertThat("Session.upgradeResponse", sess.getUpgradeResponse(), notNullValue());
+
+            Collection<WebSocketSession> sessions = client.getOpenSessions();
+            assertThat("client.sessions.size", sessions.size(), is(1));
+
+            RemoteEndpoint remote = cliSock.getSession().getRemote();
+            remote.sendPartialString("Hello", false);
+            remote.sendPartialString(" ", false);
+            remote.sendPartialString("World", true);
+
+            remote.sendPartialBytes(BufferUtil.toBuffer("It's a big enough umbrella, "), false);
+            remote.sendPartialBytes(BufferUtil.toBuffer("but it's always me that "), false);
+            remote.sendPartialBytes(BufferUtil.toBuffer("ends up getting wet."), true);
```

Comment:
seems like this is missing something, there is no test that this binary message was received

Reply:
Fixed in commit b8f29630eddd9575ad142dc9821c54a2d26f16c0